### PR TITLE
More tests

### PR DESCRIPTION
**Pull request by zagrodzki**
Includes the "coverage" branch too.

### CODE REVIEW

File: usb/fakelibusb_test.go [hunk 1]
```diff
@@ -32,14 +151,101 @@ type fakeTransfer struct {
...
 	ts map[*libusbTransfer]*fakeTransfer
 	// submitted receives a fakeTransfers when submit() is called.
 	submitted chan *fakeTransfer
+	// handles is a map of device handles pointing at opened devices.
+	handles map[*libusbDevHandle]*libusbDevice
+	// claims is a map of devices to a set of claimed interfaces
+	claims map[*libusbDevice]map[uint8]bool
+}
+
+func (f *fakeLibusb) init() (*libusbContext, error)                       { return new(libusbContext), nil }
```
Comment (kylelemons) [inline usb/fakelibusb_test.go]: Might be a good idea to throw a comment above each block of related methods explaining the fidelity of the fakes; e.g. a comment above the configs saing "Fake devices have config 1 and can only be set to that config" or whatever

Comment (zagrodzki) [inline usb/fakelibusb_test.go]: I tried to describe the behavior of the fake in the comments of fakeDevices. The implementation just follows whatever behavior is defined by the fakeDevices tree + whatever mocking mechanism the fakeLibusb provides - e.g. the devices don't have a defined behavior for transfers, but instead fakeLibusb provides waitForSubmitted which allows for custom manipulation of the transfer contents.

I could add an overall comment on the structure of fakeLibusb in the package doc. That sounds good enough?

Comment (kylelemons) [inline usb/fakelibusb_test.go]: I didn't see any comments in this diff, so if it's in the existing code nvm.

Comment (zagrodzki) [inline usb/fakelibusb_test.go]: lines 27-29 for device #1, lines 60-63 for device #2. They are very terse, so perhaps an overview of fakeLibusb at the top will be useful.

Comment (kylelemons) [inline usb/fakelibusb_test.go]: Yeah, those describe the specific fake devices, but don't describe in particular some of the behavioral interactions.  A few more words wouldn't go amiss, though realistically someone's going to have to inspect the code to write a good unit test, so I wouldn't spend too much time worrying about it.

**Pull request merged**